dsfs/dataset.go

```diff
@@ -263,6 +265,27 @@ func prepareDataset(store cafs.Filestore, ds *dataset.Dataset, df cafs.File, pri
 	return memfs.NewMemfileBytes("data."+ds.Structure.Format.String(), data), nil
 }
 
+// confirmChangesOc
+func confirmChangesOccurred(store cafs.Filestore, ds *dataset.Dataset, df cafs.File) error {
+	if ds.PreviousPath == "" {
+		return nil
+	}
+	prevKey := datastore.NewKey(ds.PreviousPath)
+	prev, err := LoadDataset(store, prevKey)
+	if err != nil {
+		return fmt.Errorf("error loading previous dataset: %s", err.Error())
+	}
+	diffMap, err := datasetDiffer.DiffDatasets(prev, ds)
+	if err != nil {
+		return err
+	}
+
+	if datasetDiffer.MapDiffsToString(diffMap) == "" {
+		return fmt.Errorf("cannot record changes if no changes occured")
```

Comment:
Not necessary right now, but in the future we should try to make error messages as clear and human-readable as possible while still being precise, But before we get too deep on error changes we should all figure out an approach & style of writing errors that works across the board. Just want to bring this to your attention now. We might also want to declare an `ErrNoChanges` variable in datasetDiffer to detect this error elsewhere.